src/Scrubber.php

```diff
@@ -70,12 +87,31 @@ protected function scrubArray(&$arr, $replacement = '*')
         
         $scrubber = $this;
 
-        $scrubberFn = function (&$val, $key) use ($fields, $replacement, &$scrubberFn, $scrubber) {
+        $scrubberFn = function (
+            &$val,
+            $key
+        ) use (
+            $fields,
+            $replacement,
+            &$scrubberFn,
+            $scrubber,
+            &$path
+) {
+        
+            $parent = $path;
+            $current = !$path ? $key : $path . '.' . $key;
+            
+            if (in_array($current, $scrubber->getWhitelist())) {
+                return;
```

Comment:
If `person` is in scrub_fields and `data.person.username` is in the whitelist, then I am pretty sure we would end up with `data.person === '********'` which I think is the correct behaviour that we want, but is not necessarily obvious. This only works at whitelisting things that would otherwise be scrubbed based on the exact key and not a earlier path component. i.e elements of the whitelist `data.a.b.c.d` will only be whitelisted if a, b, and c are not scrub_fields.